src/main/scala/beam/sim/BeamHelper.scala

```diff
@@ -284,27 +284,56 @@ trait BeamHelper extends LazyLogging {
       LinkLevelOperations.getLinkTreeMap(networkCoordinator.network.getLinks.values().asScala.toSeq)
     val linkIdMapping: Map[Id[Link], Link] = LinkLevelOperations.getLinkIdMapping(networkCoordinator.network)
     val linkToTAZMapping: Map[Link, TAZ] = LinkLevelOperations.getLinkToTazMapping(networkCoordinator.network, tazMap)
-    val freightCarriers = if (beamConfig.beam.agentsim.agents.freight.enabled) {
+    val freightConfig = beamConfig.beam.agentsim.agents.freight
+    val (freightCarriers, fixedActivitiesDurationsFromFreight) = if (freightConfig.enabled) {
+      val geoUtils = new GeoUtilsImpl(beamConfig)
       val rand: Random = new Random(beamConfig.matsim.modules.global.randomSeed)
-      val tours = PayloadPlansConverter.readFreightTours(beamConfig.beam.agentsim.agents.freight.toursFilePath)
-      val plans =
-        PayloadPlansConverter.readPayloadPlans(beamConfig.beam.agentsim.agents.freight.plansFilePath, tazMap, rand)
-      PayloadPlansConverter.readFreightCarriers(
-        beamConfig.beam.agentsim.agents.freight.carriersFilePath,
+      val tours = PayloadPlansConverter.readFreightTours(
+        freightConfig,
+        geoUtils,
+        networkCoordinator.transportNetwork.streetLayer
+      )
+      val plans = PayloadPlansConverter.readPayloadPlans(
+        freightConfig,
+        geoUtils,
+        networkCoordinator.transportNetwork.streetLayer
+      )
+      val carriers = PayloadPlansConverter.readFreightCarriers(
+        freightConfig,
+        geoUtils,
+        networkCoordinator.transportNetwork.streetLayer,
         tours,
         plans,
         vehicleTypes,
-        tazMap,
         rand
```

Comment:
I think this code is big enough to extract it to some method with a meaningful method name.